libdnf/rpm/solv_sack.cpp

```diff
@@ -349,9 +350,60 @@ void SolvSack::Impl::internalize_libsolv_repos() {
     }
 }
 
-void SolvSack::Impl::load_repo(Repo & repo, bool build_cache, LoadRepoFlags flags) {
+bool SolvSack::Impl::load_repo(Repo & repo, bool build_cache, LoadRepoFlags flags) {
+    auto repo_impl = repo.p_impl.get();
+
+    switch (repo_impl->type) {
+        case Repo::Type::SYSTEM:
+            return load_system_repo(repo);
+        case Repo::Type::AVAILABLE:
+            load_available_repo(repo, build_cache, flags);
+            return true;
+        case Repo::Type::COMMANDLINE:
+            // TODO(jrohel): Implement command line repo loading
+            throw LogicError("Not implemented");
+    }
+
+    return false;
+}
+
+bool SolvSack::Impl::load_system_repo(Repo & repo) {
+    auto & logger = base->get_logger();
+    auto repo_impl = repo.p_impl.get();
+    auto id = repo.get_id().c_str();
+
+    std::unique_ptr<LibsolvRepo, decltype(&libsolv_repo_free)> libsolv_repo(repo_create(pool, id), &libsolv_repo_free);
+
+    logger.debug("load_system_repo(): fetching rpmdb");
+    int flagsrpm = REPO_REUSE_REPODATA | RPM_ADD_WITH_HDRID | REPO_USE_ROOTDIR;
+    int rc = repo_add_rpmdb(libsolv_repo.get(), nullptr, flagsrpm);
+    if (rc != 0) {
+        // g_set_error(error, DNF_ERROR, DNF_ERROR_FILE_INVALID, _("failed loading RPMDB"));
+        logger.warning(fmt::format(_("load_system_repo(): failed loading RPMDB: {}"), pool_errstr(pool)));
+        return false;
+    }
+
+    repo_impl->attach_libsolv_repo(libsolv_repo.release());
```

Comment:
This step makes a test that there is no libsolv repo attached. What about to do the test before the new libsolv repo is created?

Reply:
> This step makes a test that there is no libsolv repo attached.

Attached libsolv repository indicates a program error. The test throws LogicError and program will be stopped (traceback?). Performing a test before a new libsolv repository is created does not improve the situation.

Another question is whether we want to change the behaviour of the `load_system_repo()` to support reloading. But usually there are the "available" repositories behind the "system" repository. So I'm not sure if reloading the "system" repository (e.g. after transaction) without reseting SolvSack is a good idea.